### ISSUE TITLE
FATAL {agency} Assumed active RAFT peer has no active agency list: Expecting type DoubleAdministrative intervention needed.

### ISSUE THREAD

**Issue by brucejee** [arangodb/arangodb#2238]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [ ] 2.8
- [ ] 3.0
- [x] 3.1.6
- [ ] self-compiled devel branch

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ ] own infrastructure
- [ ] Linux 
  - [ ] Debian .deb
  - [ ] Ubuntu .deb
  - [ ] SUSE   .rpm
  - [ ] RedHat .rpm
  - [ ] Fedora .rpm
  - [ ] Gentoo
  - [x] docker - official docker library
  - [ ] other:
- [ ] Windows, version: 
- [ ] MacOS, version:

### problem

Coordinator have more logs like this, and i try to restart the agency, and receive the following error.

#### Cooridnator log:
2016-12-27T03:27:43Z [1] WARNING {agencycomm} Waiting on leader election http+tcp://192.168.1.81:15001 tries: 134
2016-12-27T03:27:44Z [1] WARNING {agencycomm} Waiting on leader election http+tcp://192.168.1.81:15001 tries: 136
2016-12-27T03:27:44Z [1] WARNING {agencycomm} Waiting on leader election http+tcp://192.168.1.81:15001 tries: 136
2016-12-27T03:27:45Z [1] WARNING {agencycomm} Waiting on leader election http+tcp://192.168.1.81:15001 tries: 135
2016-12-27T03:27:47Z [1] WARNING {agencycomm} Waiting on leader election http+tcp://192.168.1.81:15001 tries: 137
2016-12-27T03:27:47Z [1] WARNING {agencycomm} Waiting on leader election http+tcp://192.168.1.81:15001 tries: 137
2016-12-27T03:27:48Z [1] WARNING {agencycomm} Waiting on leader election http+tcp://192.168.1.81:15001 tries: 136

#### Restart the agency, and receive this error:
016-12-27T03:32:19Z [1] INFO ArangoDB 3.1.6 [linux] 64bit, using VPack 0.1.30, ICU 54.1, V8 5.0.71.39, OpenSSL 1.0.1t  3 May 2016
2016-12-27T03:32:19Z [1] INFO using SSL options: SSL_OP_CIPHER_SERVER_PREFERENCE, SSL_OP_TLS_ROLLBACK_BUG
2016-12-27T03:32:19Z [1] INFO file-descriptors (nofiles) hard limit is 65536, soft limit is 65536
2016-12-27T03:32:19Z [1] INFO running WAL recovery (1 logfiles)
2016-12-27T03:32:19Z [1] INFO replaying WAL logfile '/var/lib/arangodb3/journals/logfile-8400644.db' (1 of 1)
2016-12-27T03:32:19Z [1] INFO WAL recovery finished successfully
2016-12-27T03:32:19Z [1] INFO JavaScript using startup '/usr/share/arangodb3/js', application '/var/lib/arangodb3-apps'
2016-12-27T03:32:19Z [1] INFO In database '_system': Database is up-to-date (30106/standalone/existing)
2016-12-27T03:32:19Z [1] INFO using endpoint 'http+tcp://0.0.0.0:5001' for non-encrypted requests
2016-12-27T03:32:20Z [1] INFO ArangoDB (version 3.1.6 [linux]) is ready for business. Have fun!
2016-12-27T03:32:21Z [1] INFO {agency} Restarting agent from persistence ...
2016-12-27T03:32:21Z [1] INFO {agency} Found active RAFTing agency lead by f94eadba-7fe7-4a3e-bc47-05fdf80ae213. Finishing startup sequence.
2016-12-27T03:32:21Z [1] FATAL {agency} Assumed active RAFT peer has no active agency list: Expecting type DoubleAdministrative intervention needed.

**Comment by brucejee** [comment]
I stop all agency, and restart agency. The coordinator don't log the error again.
But after i import more data, the coordinator log the same following error again.

The only way to fix this error is restart the three agency.

### coordinator log
2016-12-27T06:45:33Z [1] WARNING {agencycomm} Waiting on leader election http+tcp://172.16.10.7:15001 tries: 378
2016-12-27T06:45:34Z [1] WARNING {agencycomm} Waiting on leader election http+tcp://172.16.10.7:15001 tries: 365
2016-12-27T06:45:34Z [1] WARNING {agencycomm} Waiting on leader election http+tcp://172.16.10.7:15001 tries: 374
2016-12-27T06:45:34Z [1] WARNING {agencycomm} Waiting on leader election http+tcp://192.168.1.81:15001 tries: 363
2016-12-27T06:45:34Z [1] WARNING {agencycomm} Waiting on leader election http+tcp://172.16.10.7:15001 tries: 379

### agency log
2016-12-27T06:09:57Z [1] WARNING {agency} Not accepting appendEntries from f94eadba-7fe7-4a3e-bc47-05fdf80ae213
2016-12-27T06:09:58Z [1] WARNING {agency} Not accepting appendEntries from f94eadba-7fe7-4a3e-bc47-05fdf80ae213
2016-12-27T06:09:58Z [1] WARNING {agency} Not accepting appendEntries from f94eadba-7fe7-4a3e-bc47-05fdf80ae213

**Comment by brucejee** [comment]
When i want to test CRUD operations, coordinator again log the error, and restart agency can fix this. 
2016-12-29T05:20:28Z [1] WARNING {agencycomm} Waiting on leader election http+tcp://172.16.10.7:15002 tries: 3062
2016-12-29T05:20:29Z [1] WARNING {agencycomm} Waiting on leader election http+tcp://172.16.10.7:15002 tries: 3059
2016-12-29T05:20:29Z [1] WARNING {agencycomm} Waiting on leader election http+tcp://172.16.10.7:15002 tries: 3039
2016-12-29T05:20:29Z [1] WARNING {agencycomm} Waiting on leader election http+tcp://192.168.1.81:15001 tries: 2958

**Comment by brucejee** [comment]
But arise another error now:

2016-12-29T08:15:55Z [1] ERROR {cluster} ClusterComm::performRequests: got no answer from shard:s100083:/_db/KG_DB/_api/document?collection=s100083&waitForSync=false&returnNew=false&returnOld=false with error 4
2016-12-29T08:16:15Z [1] ERROR {cluster} ClusterComm::performRequests: got no answer from shard:s100083:/_db/KG_DB/_api/document?collection=s100083&waitForSync=false&returnNew=false&returnOld=false with error 4
2016-12-29T08:16:15Z [1] ERROR {cluster} ClusterComm::performRequests: got no answer from shard:s100083:/_db/KG_DB/_api/document?collection=s100083&waitForSync=false&returnNew=false&returnOld=false with error 4
2016-12-29T08:16:16Z [1] ERROR {cluster} ClusterComm::performRequests: got no answer from shard:s100083:/_db/KG_DB/_api/document?collection=s100083&waitForSync=false&returnNew=false&returnOld=false with error 4
2016-12-29T08:16:16Z [1] ERROR {cluster} ClusterComm::performRequests: got no answer from shard:s100083:/_db/KG_DB/_api/document?collection=s100083&waitForSync=false&returnNew=false&returnOld=false with error 4
2016-12-29T08:16:17Z [1] ERROR {cluster} ClusterComm::performRequests: got no answer from shard:s100083:/_db/KG_DB/_api/document?collection=s100083&waitForSync=false&returnNew=false&returnOld=false with error 4
2016-12-29T08:16:17Z [1] ERROR {cluster} ClusterComm::performRequests: got no answer from shard:s100083:/_db/KG_DB/_api/document?collection=s100083&waitForSync=false&returnNew=false&returnOld=false with error 4

**Comment by kvahed** [comment]
we would need a little more information on setup and startup to be able to help you here. and complete logs will only reveal the intermediate problems that came up. it looks that a lot has gone wrong above and we are seeing only a tiny window on the problems.

**Comment by kvahed** [comment]
nevermind i have found the bug. i have one bug that i could associate with your above logs. thanks. will be fixed with the next bug fix release.

**Comment by brucejee** [comment]
My setup is three agencies, two coordinators, two primary DB server.
![image](https://cloud.githubusercontent.com/assets/5820173/21580225/383047f4-d020-11e6-9086-de52e6c55063.png)

When I import 2 million docs, it arise  "Waiting on leader election" error. And i only use one agency to avoid this error.

But this error may arise when i import docs.

2016-12-29T08:15:55Z [1] ERROR {cluster} ClusterComm::performRequests: got no answer from shard:s100083:/_db/KG_DB/_api/document?collection=s100083&waitForSync=false&returnNew=false&returnOld=false with error 4

**Comment by kvahed** [comment]
We are in the process of releasing 3.1.8 containing among others this bug's fix. Thx again.

**Comment by brucejee** [comment]
The following error also exist in docker 3.1.8 image when i import 1 million docs use Hadoop mappers.

2017-01-12T04:06:50Z [1] ERROR {cluster} ClusterComm::performRequests: got no answer from shard:s3100407:/_db/KG_DB/_api/document?collection=s3100407&waitForSync=false&returnNew=false&returnOld=false with error 4
2017-01-12T04:07:01Z [1] ERROR {cluster} ClusterComm::performRequests: got no answer from shard:s3100407:/_db/KG_DB/_api/document?collection=s3100407&waitForSync=false&returnNew=false&returnOld=false with error 4
2017-01-12T04:07:01Z [1] ERROR {cluster} ClusterComm::performRequests: got no answer from shard:s3100406:/_db/KG_DB/_api/document?collection=s3100406&waitForSync=false&returnNew=false&returnOld=false with error 4
2017-01-12T04:07:02Z [1] ERROR {cluster} ClusterComm::performRequests: got no answer from shard:s100018:/_db/_system/_api/collection/s100018/count with error 4

**Comment by m0ppers** [comment]
I have ported your problem over to a new issue here: #2265

The agency problems seem to have gone? This is a new problem so I am closing this one. If you still have agency problems related to this issue please reopen!

**Comment by brucejee** [comment]
Thank for porting, waiting for this problem fixed.
